### PR TITLE
Component Unit Tests

### PR DESCRIPTION
**Pull request by bensquire**
@adamjimenez These tests cover all the low hanging fruit. i.e component code that doesn't use the DB.

It would probably be useful to go through the tests and check the assertions; I've generally assumed that whats there is correct, but I mght have written the tests in such a way that highlights issues.

### CODE REVIEW

File: _lib/cms/components/AvgRating.php [hunk 1]
```diff
@@ -22,9 +22,11 @@ public function getFieldSql(): ?string
...
-        return '<select name="' . $fieldName . '" class="rating" data-section="' . $this->cms->section . '" data-item="' . $this->cms->content['id'] . '"  data-avg="data-avg" ' . $options['attribs'] . '>
-            <option value="">Choose</option>
-            ' . html_options(self::RATING_OPTS, $value, true) . '
-        </select>';
+        $html = [];
+        $html[] = '<select name="' . $fieldName . '" class="rating" data-section="' . $this->cms->getSection() . '" data-item="' . $this->cms->getContent()['id'] . '" data-avg="data-avg" ' . $options['attribs'] . '>';
+        $html[] = '<option value="">Choose</option>';
+        $html[] = html_options(self::RATING_OPTS, $value, true);
+        $html[] = '</select>';
+        return implode('', $html);
```
Comment (adamjimenez) [inline _lib/cms/components/AvgRating.php]: why not concatenate instead of creating an array and imploding?

**Comment by adamjimenez** [comment]
Great to see a unit test!
Probably the most critical parts for unit tests will be making sure cms get and save works correctly, which will require a test database.

File: _lib/cms/components/AvgRating.php (hunk 1 above)
Comment (bensquire) [reply _lib/cms/components/AvgRating.php]: I think it just looks cleaner, but I've updated the code either way

**Comment by bensquire** [comment]
@adamjimenez Is it just me or is there an error in:

`_lib/cms/Component.php:conditionsToSql` ?

Shouldn't

`return $tablePrefix . $fieldName . " LIKE '" . escape($value) . "'";`

be 

`return $tablePrefix . '.' . $fieldName . " LIKE '" . escape($value) . "'";`

File: _lib/core/auth.php [hunk 2]
```diff
@@ -292,6 +300,8 @@ public function load()
      */
     public function create_hash(string $password): string
     {
+        var_dump($this->hash_password ? hash('sha256', $this->hash_salt . $password) : $password);
+        die('here2');
```
Comment (adamjimenez) [inline _lib/core/auth.php]: hmm

Comment (bensquire) [reply _lib/core/auth.php]: Good catch. Fixed.

**Comment by adamjimenez** [comment]
> @adamjimenez Is it just me or is there an error in:
> 
> `_lib/cms/Component.php:conditionsToSql` ?
> 
> Shouldn't
> 
> `return $tablePrefix . $fieldName . " LIKE '" . escape($value) . "'";`
> 
> be
> 
> `return $tablePrefix . '.' . $fieldName . " LIKE '" . escape($value) . "'";`

The dot comes with:
https://github.com/adamjimenez/shiftlib/blob/master/_lib/cms/cms.php#L348

**Pull request merged**